xtask/src/ci.rs

```diff
@@ -95,12 +98,11 @@ impl CiTask {
                 --order package,lib,features,dependencies,dev-dependencies,build-dependencies
             "
         )
-        .run();
+        .run()
+        .map_err(Into::into);
+        // Check that all links point to the same version of the spec
+        let spec_links = check_spec_links(&self.project_root.join("crates"));
 
-        println!("Checking all Matrix Spec links point to same version...");
-        check_spec_links(&self.project_root.join("crates"))?;
-        println!("All links to the Matrix Spec are consistent!");
-
-        fmt_res.and(check_full_res).and(clippy_res).and(sort_res).map_err(Into::into)
+        fmt_res.and(check_full_res).and(clippy_res).and(sort_res).and(spec_links)
```

Comment:
You can remove all the previous `.map_err`s if you add one before `.and(spec_links)` here I think.

Reply:
No `Box<dyn Error>` can't be converted to `xshell::Error` :cry: